docker.Makefile

```diff
@@ -110,9 +110,19 @@ lint: ## run linter(s)
 
 vendor: build_dev_image
 	$(info Update Vendoring...)
-	docker run --rm -v "$(CURDIR)":/go/src/github.com/docker/app/ $(DEV_IMAGE_NAME) make vendor
+	rm -rf ./vendor
+	docker rm -f docker-app-vendoring || true
+	docker create --name docker-app-vendoring -v docker-app-vendor-cache://dep-cache -e DEPCACHEDIR=//dep-cache $(DEV_IMAGE_NAME) sh -c "rm -rf ./vendor && make vendor"
+	docker start -i -a docker-app-vendoring
```

Comment:
the container doesn't have to be running to use `docker cp`

Reply:
Oh, nvm, this `start` triggers the actual vendoring, right?

An alternative would be to do the vendoring in `docker build` (in a build-stage). If you're using buildkit, you could even use a `RUN --mount`

Reply:
See https://github.com/moby/buildkit/blob/master/hack/update-vendor for an example

Reply:
I think we aren't quite ready (in CI at least, if not on contributors machines) to mandate buildkit.

Reply:
Hm, yeah. And with the legacy builder, the caching would be written to the container's filesystem (so likely slower).

Still curious why `docker create` and `docker start` are done separately; could just `docker run` the container (and have it exit afterwards)

Reply:
Yes, I removed the create and collapsed everything in start (in upcoming commit)